bundle/deploy/terraform/state_push.go

```diff
@@ -30,6 +32,12 @@ func (l *statePush) Apply(ctx context.Context, b *bundle.Bundle) error {
 	// Expect the state file to live under dir.
 	local, err := os.Open(filepath.Join(dir, TerraformStateFileName))
 	if err != nil {
+		if errors.Is(err, fs.ErrNotExist) && b.TerraformHasNoResources {
+			// A terraform state file is not created for new bundle deployments
+			// with no resources defined. We ignore that a local terraform state
+			// file is absent if the bundle config has no resources defined.
+			return nil
+		}
```

Comment:
Please add an integration test for this. I imagine 2 types:
1. One where there are no resources to begin with
2. One where there initially is a single job (for ex), a deploy, removal of the job, then another deploy

Both cases should pass, but in the second case we still need to upload the state file.

Reply:
done